### DISCUSSION TITLE
Add a badge to indicate whether a crate supports static linking

### DISCUSSION THREAD

**Discussion by adaszko** [rust-lang/crates.io#4802] [Feature Requests]
Hi!

This is related to https://github.com/rust-lang/crates.io/issues/737 in that both make it easier to quickly filter relevant crates.

Static linking makes it much easier to distribute binaries.  Either in the form of only executable files or as a .deb package or a Homebrew formula.  To minimize additional problems caused by not shipping/shipping the wrong shared libraries, it's best to simply link the binary statically.  For that to happen, every crate that the project uses needs to support linking statically itself.  And that's where this issue/discussion comes in.  When looking at a crate, it's currently non-obvious whether it supports static linking or not hence the badge.

It would potentially be misleading to implement that badge just in a declarative manner: the crates would probably need to be somehow automatically checked per-release whether it links statically successfully.  Perhaps by tying somehow with the CI system of the crate.  Any thoughts? :)

**Comment by Turbo87** [comment]
it would certainly be nice to know, but hard to do in practice. if a crate depends on another one and that other one suddenly changes its "does compile statically" status, then the first one would also have to change the status and we would get this cascade across the ecosystem. I'm not sure how we could currently build this in a somewhat scalable way.
